### PR TITLE
Add option to send kudos from profile

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

1. Users now seeing a button to send kudos to this user:
image
2. When the sender user clicks the button, is redirected to the page with the user to field populated with the user info:
image

##### Refers/Fixes

#5661

### CODE REVIEW

**Comment by owocki** [comment]
thanks; does the kudos stick as they click 'pick a kudos' and browse the marketplace too?

**Comment by owocki** [comment]
@zoek1 note that this PR differs slightly from the expected behaviour in the ticket in that the kudos to username specification will be lost when the user browses to select their kudos.. it should probably stick until cleared.

**Comment by zoek1** [comment]
Yep, I missed that. I'm changing that to persist the username with this rules:
1. Until the kudos is sent, clear persistent username
2. Until the user removes the username manually and for a max period of one day. In such a case, the stored username will be discarded.

@owocki

**Comment by zoek1** [comment]
The changes are ready @owocki, maybe three hours could be better instead of one day to remove the receiver user from local storage, what do you think?

**Comment by owocki** [comment]
@zoek1 i think 3 hours is fine.. also if the sending user clicks out of that user; then i think that should remove it too.

**Comment by zoek1** [comment]
By click out what do you mean?  when sending user? As I see you refer to some of these:
1. when sending user select another user from the list or 
2. select another tab to send kudos (eth address, anywhere)

**Comment by owocki** [comment]
@zoek1 both

**Comment by zoek1** [comment]
@owocki the users now is removed when:
> 1. When sender user selects another user from the list or
> 2. when sender user selects another tab to send kudos (eth address, anywhere)

File: app/kudos/templates/transaction/send.html [hunk 1]
```diff
@@ -165,4 +173,50 @@ <h3 class="font-weight-300">{% blocktrans %}<span id="new_username">the user</sp
...
+    let data = JSON.parse(encoded_data);
+    let current = new Date();
+    let past = new Date(Date.parse(data.date))
+
+    if ((current.getTime() - past.getTime()) / (60 * 60) > 1440) {
+      localStorage.removeItem('current_receiver')
+    } else {
+      let option = document.createElement('option')
+      option.value = data.id
+      option.setAttribute('avatar_id',  data.avatar_id)
```
Comment (danlipert) [inline app/kudos/templates/transaction/send.html]: Theres a couple places in this PR with some spacing issues - can you take another look and clean these up? thanks!

**Comment by zoek1** [comment]
Sure, i'll check and fix it!

File: app/kudos/views.py [hunk 2]
```diff
@@ -289,6 +289,22 @@ def send_2(request):
...
         raise Http404
 
+    username = request.GET.get('username', None)
+    user = {}
+
+    if username:
+        profiles = Profile.objects.filter(handle__iexact=username)
+
+        if profiles.exists():
+            profile = profiles.first()
```
Comment (danlipert) [inline app/kudos/views.py]: If we had two users with handles "zoek1" and "zoek1asdfasdf", and I wanted to send a kudos to "zoek1asdfasdf", would this work?

Comment (zoek1) [reply app/kudos/views.py]: Sure, It works because the filter method uses iexact on the handle attribute, in the previous bounty you recommended that

**Comment by owocki** [comment]
pumped to get this live.. (once it passes code review).. seems like its close to me

**Comment by zoek1** [comment]
@owocki also this ready!

**Comment by zoek1** [comment]
fix conflicts :+1:

**Comment by owocki** [comment]
pumped for this

File: app/dashboard/templates/profiles/header_details.html [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {% load i18n static avatar_tags add_url_schema email_obfuscator %}
 
 <h1 class="profile-header__handle">
-  @{{ profile.name }}
```
Comment (danlipert) [inline app/dashboard/templates/profiles/header_details.html]: Was this removed by mistake?

Comment (zoek1) [reply app/dashboard/templates/profiles/header_details.html]: :see_no_evil: Thanks, yes deleted by mistake. I just restored this line.

**Pull request merged**